R-package/R/lgb.Booster.R

```diff
@@ -474,14 +498,43 @@ Booster <- R6::R6Class(
         num_iteration <- self$best_iter
       }
 
-      # Return model string
-      return(
-        lgb.call.return.str(
-          fun_name = "LGBM_BoosterSaveModelToString_R"
+      # Create buffer
```

Comment:
Maybe write a new helper function for calls with buffer allocation? I believe it will improve the code readability.

Reply:
That was exactly what `lgb.call.return.str()` was. But since you cannot pass a string with a function name to `.Call()` once registration is used (https://github.com/microsoft/LightGBM/pull/4155#issuecomment-812315498), I couldn't see a good way to keep a shared helper function.

I also think we'll be able to factor this code away completely soon, as part of #3016, by just creating the R character vector on the C++ side and returning it. See #4242 for a first example of that.

Reply:
Ah, OK, I see!